presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2183,6 +2194,62 @@ public void dropView(ConnectorSession session, SchemaTableName viewName)
         return views.build();
     }
 
+    @Override
+    public Optional<ConnectorMaterializedViewDefinition> getMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        requireNonNull(viewName, "viewName is null");
+        Optional<Table> table = metastore.getTable(viewName.getSchemaName(), viewName.getTableName());
+        if (table.isPresent() && MetastoreUtil.isPrestoMaterializedView(table.get())) {
+            try {
+                JsonCodec<ConnectorMaterializedViewDefinition> codec = JsonCodec.jsonCodec(ConnectorMaterializedViewDefinition.class);
+                return Optional.of(codec.fromJson(decodeViewData(table.get().getViewOriginalText().get())));
```

Comment:
It's all about, at the level of HiveMetadata, we need it to be a deserialized structure, so that we could get the list of base tables out of it and update the MV links kept in base table. So an already serialized string doesn't work here.

Also we need to include the list of base tables in the serialized structure and save it to Metastore when creating MV, so that when dropping MV we could retrieve the base tables, again at the level of HiveMetadata.